bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -893,67 +1078,66 @@ private void setSelectionFromNode() {
         final TargetObsComp dataObject = getDataObject();
         if ((obsComponent == null) || (dataObject == null)) return;
 
+        // We don't allow this for coordinates.
         final Option<SPTarget> targetOpt = TargetSelection.getTargetForNode(dataObject.getTargetEnvironment(), obsComponent);
-        if (targetOpt.isDefined()) {
-            targetOpt.foreach(target -> {
-
-                // Clone the target.
-                final SPTarget newTarget = new SPTarget();
-                newTarget.setTarget(target.getTarget());
-
-                // Add it to the environment.  First we have to figure out what it is.
-                final TargetEnvironment env = dataObject.getTargetEnvironment();
-
-                // See if it is a guide star and duplicate it in the correct GuideTargets list.
-                boolean duplicated = false;
-                env.getPrimaryGuideGroup();
-                final List<GuideGroup> groups = new ArrayList<>();
-                for (GuideGroup group : env.getGroups()) {
-                    for (GuideProbeTargets gt : group) {
-                        if (gt.getTargets().contains(target)) {
-                            group = group.put(gt.update(OptionsList.UpdateOps.append(newTarget)));
-                            duplicated = true;
-                            break;
-                        }
+        final Option<IndexedGuideGroup> iggOpt =
+                TargetSelection.getIndexedGuideGroupForNode(dataObject.getTargetEnvironment(), obsComponent);
+        targetOpt.foreach(target -> {
```

Comment:
The target vs. group cases used to be in an `if / then/ else` statement.  Now each will be executed if their respective `Option`s are defined.  Is it the case that if one is defined the other definitely won't be?

Reply:
If more than one is defined, then something has definitely gone very wrong in the OT.

I have no opposition to switching to and `if` block if you think it's safer, but realistically, if more of one of these executes, we should bork because the internal state is messed up.

Reply:
> realistically, if more of one of these executes, we should bork because the internal state is messed up.
Perhaps we should make that explicit and throw an exception in that case?

Reply:
Sounds good. I'll change that.

Reply:
(Actually, this is a very good idea, because during development, I reached a point where I was creating things incorrectly, and I did have more than one thing set per node.)

I have added a `checkTargetSelectionConsistency` method that checks to make sure 0 or 1 things are selected, and otherwise throws a `RuntimeException`. I call it before any `TargetSelection` access as we have here. (It'll be in my next commit.)